.github/workflows/test.yml

```diff
@@ -59,47 +60,60 @@ jobs:
       PYTHONDEVMODE: '1'
     steps:
       - name: Set up Python
-        uses: actions/setup-python@v5
+        uses: Quansight-Labs/setup-python@v5
```

Comment:
Why do we need to adopt a less official action? Does `actions/setup-python` not support the free-threaded builds?

Reply:
Unfortunately, GitHub has been quite unresponsive to this: https://github.com/actions/setup-python/issues/771

Though, there seems to be some recent activity: https://github.com/actions/setup-python/pull/973

Reply:
The fork is tracking upstream and has the open PR to add free-threading support applied.

You could also use `setup-uv`, which can be used as a drop-in replacement if you install pip into the uv environment.

Reply:
Well, my (former) project provides the Python builds for uv. I actually trust those Python builds and the people behind uv. So astral-sh/setup-uv would be my preference.